### PR TITLE
Complete Graphql update to v14

### LINKED ISSUE #11507: Please update express-graphql@0.6.12 and graphql-tools@3.1.1 eslint-plugin-graphql@2.1.1 warns about peer dependency graphql
If I'm `npm i` my app with `graphql@0.13.2` I have this dependency issues with gatsby sub-dependencies:
```
npm WARN eslint-plugin-graphql@2.1.1 requires a peer of graphql@^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
npm WARN babel-plugin-remove-graphql-queries@2.5.4 requiresa peer of graphql@^14.1.1 but none is installed. You must install peer dependencies yourself.
npm WARN graphql-skip-limit@2.0.5 requires a peer of graphql@^14.1.1 but none is installed. You must install peer dependencies yourself.
npm WARN relay-compiler@2.0.0 requires a peer of graphql@^14.1.0 but none is installed. You must install peer dependencies yourself.
```

if I'm `npm i` my app with `graphql@14.1.1` I have next sub-dependency issues
```
npm WARN express-graphql@0.6.12 requires a peer of graphql@^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-plugin-graphql@2.1.1 requires a peer of graphql@^0.12.0 || ^0.13.0 but none is installed. You must install peer dependencies yourself.
npm WARN graphql-tools@3.1.1 requires a peer of graphql@^0.13.0 but none is installed. You must install peer dependencies yourself.
```
current versions of this dependencies: 
express-graphql@0.7.1 - safe to update
graphql-tools@4.0.4  - safe to update
eslint-plugin-graphql@3.0.3 - need to be tested

### PR DESCRIPTION
**Pull request by sibiraj-s**
## Description

Recently Graphql is updated to v14. this should complete the update.

**Note**: I have updated the package `documentation` in `gatsby-transformer-documentation` as jest tests failed due to some issue is caused by one of its dependencies during post install. the package updated from `v7` to `v9` which is breaking.

## Related Issues

Fixes #11507

### CODE REVIEW

**Comment by saranshkataria** [comment]
@sibiraj-s  I messed up while resolving the conflicts and now cannot push to the branch to fix it, sorry about that!

**Comment by pieh** [comment]
> **Note**: I have updated the package `documentation` in `gatsby-transformer-documentation` as jest tests failed due to some issue is caused by one of its dependencies during post install. the package updated from `v7` to `v9` which is breaking.

This is super weird - just for test I'll temporarily remove `documentation` change (and fixup current problem with `yarn.lock` ;P). Ideally we wouldn't update unrelated stuff here - I could bump `documentation` package as part of https://github.com/gatsbyjs/gatsby/pull/11597

**Comment by pieh** [comment]
@saranshkataria when there are conflicts in yarn.lock file you can just run `yarn [bootstrap` and it will resolve conflicts automatically, you don't need to resolve them manually (and it's also super difficult to do in general) :)

**Comment by pieh** [comment]
@sibiraj-s it seems only failing test is caused by some regression introduced in https://github.com/gatsbyjs/gatsby/pull/11581, so we should be good dropping unrelated `documentation` version bump (but I will do this in my PR)

File: packages/gatsby/package.json [hunk 1]
```diff
@@ -50,11 +50,12 @@
...
     "eslint-loader": "^2.1.0",
     "eslint-plugin-flowtype": "^2.46.1",
-    "eslint-plugin-graphql": "^2.0.0",
+    "eslint-plugin-graphql": "^3.0.3",
     "eslint-plugin-import": "^2.9.0",
     "eslint-plugin-jsx-a11y": "^6.0.3",
     "eslint-plugin-react": "^7.8.2",
     "express": "^4.16.3",
+    "express-graphql": "^0.7.1",
     "express-graphql": "^0.6.12",
```
Comment (akselinurmio) [inline packages/gatsby/package.json]: Have you noticed that the `express-graphql` is set twice in the dependencies? It's not currently using the version `^0.7.1` you have set, but instead the `^0.6.12`.

Comment (sibiraj-s) [reply packages/gatsby/package.json]: didn't notice will fix that..

**Comment by sibiraj-s** [comment]
> so we should be good dropping unrelated `documentation` version

Okay @pieh.

**Comment by stefanprobst** [comment]
Looks like this is the same as  #11512

**Comment by wardpeet** [comment]
Thanks so much for the PR!

Unfortunately, we're not going to merge this at this time. It seems like a duplicate of this one #11512 got reverted.

As such--we're going to close this out. We want you to know that we certainly want to keep you as a contributor to Gatsby, so we'd encourage you to check out our open issues.

Thanks again, and we look forward to seeing more PRs from you in the future! 💪 💜

**Pull request closed without merging**